packages/protoplugin-test/src/custom-options.test.ts

```diff
@@ -0,0 +1,298 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { getDescriptorSet } from "./helpers.js";
+import type { DescFile } from "@bufbuild/protobuf";
+import {
+  getCustomOptionBoolean,
+  getCustomOptionFloat,
+  getCustomOptionUint32,
+  getCustomOptionInt32,
+  getCustomOptionInt64,
+  getCustomOptionString,
+  getCustomOptionMessage,
+  getCustomOptionEnum,
+} from "@bufbuild/protoplugin/ecmascript";
+import { proto3 } from "@bufbuild/protobuf";
+
+// Used for custom options of type FooMessage
+const foo = proto3.makeMessageType("FooMessage", [
+  {
+    no: 1,
+    name: "foo",
+    kind: "scalar",
+    T: 5 /* ScalarType.INT32 */,
+  },
+  {
+    no: 2,
+    name: "bar",
+    kind: "scalar",
+    T: 9 /* ScalarType.STRING */,
+  },
+  {
+    no: 3,
+    name: "quux",
+    kind: "scalar",
+    T: 9 /* ScalarType.STRING */,
+  },
+  {
+    no: 4,
+    name: "many",
+    kind: "scalar",
+    repeated: true,
+    T: 9 /* ScalarType.STRING */,
+  },
+  {
+    no: 5,
+    name: "mapping",
+    kind: "map",
+    K: 9 /* ScalarType.STRING */,
+    V: {
+      kind: "scalar",
+      T: 9 /* ScalarType.STRING */,
+    },
+  },
+  {
+    no: 6,
+    name: "unused",
+    kind: "scalar",
+    T: 9 /* ScalarType.STRING */,
+  },
+]);
+
+describe("custom options", function () {
+  let files: DescFile[] = [];
+  beforeEach(() => {
+    files = getDescriptorSet().files.filter(
+      (file) => file.name === "proto/custom_options"
+    );
+  });
+  describe("finds options correctly", function () {
+    test("file options", () => {
+      for (const file of files) {
+        expect(getCustomOptionString(file, 50000)).toEqual("Hello");
+      }
+    });
+    test("enum options", () => {
+      for (const file of files) {
+        for (const enumeration of file.enums) {
```

Comment:
Iterating over the files is going to be brittle: If you add another proto file to packages/protoplugin/test/proto, these tests will fail. 

What do you think about changing the approach a bit, and look up an enumeration by name? `descriptorSet.enums.get("foo")` This also works for messages, files, etc.

Reply:
I am actually filtering out only the `custom_options` proto in the `beforeEach` currently, so that _should_ protect against tests failing if another file is added right?

Regardless, I'm totally cool with changing them to access the internal maps, but just wanted to see if my assumption was true.

Reply:
Sorry, I missed the filter. Still makes sense to me to be very explicit about picking something from the descriptor set, it will make it much easier to add cases if needed.

For example:

```ts
      const enumeration = descriptorSet.enums.get("FooEnum");
      assert(enumeration);
      expect(getCustomOptionBoolean(enumeration, 50004)).toBeTruthy();

      const enumValue = descriptorSet.enums.get("FooEnum")?.values.find(v => v.name === "OFF");
      assert(enumValue);
      expect(getCustomOptionBoolean(enumValue, 50004)).toBeTruthy();
```

Where `assert()` could be copied from assert.ts into a test helper. If the assertion fails, test data is missing - the tests are broken. If the expectation fails, the tests are good, but the tested code failed the test.